### PR TITLE
Make system ready for deployment

### PR DESCRIPTION
**Pull request by vgeddes**
Changes:
* Relayer
  * Added `global.data-dir` config field to set data directory for any storage
  * ethash cache location is now in the above data dir (also depends on changes in the ethashproof repo)
  * Renamed environment variables for secrets, for clarity
  * Misc improvements for handling & configuration of beefy sqlite DB
  * Parachain ID is queried from chain rather than being hardcoded.
* Parachain
  * Misc simplifications and improvements to chain spec code.
* Ethereum
  * Truffle migration script now dumps all deployed contract addresses to `/tmp/addresses.json`. This file is required by tooling in the [snowbridge-infra](https://github.com/Snowfork/snowbridge-infra) repo.
* E2E tests
  * Parachain ID now defaults to `1000` in line with guidelines (non-system parachains begin at 1000).
* Infra
  * Deleted all obsolete Ansible stuff. The new Nomad tooling lives in the [snowbridge-infra](https://github.com/Snowfork/snowbridge-infra) repo.

### CODE REVIEW

File: ethereum/migrations/4_dump_address.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require("dotenv").config();
```
Comment (musnit) [inline ethereum/migrations/4_dump_address.js:1]: re filename: 4_ is used, can we make this 6_ or keep it as 4_ and bump the others up by one?

File: parachain/scripts/make-snowbridge-image.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        echo "error: unknown parameter"
+        exit 1
+        ;;
+  esac
+done
+
+version=$(git describe --tags $(git rev-list --tags --max-count=1) | sed s/v//)
+revision=$(git rev-parse HEAD)
+
+c=$(buildah from ubuntu:20.04)
```
Comment (musnit) [inline parachain/scripts/make-snowbridge-image.sh:23]: is buildah nativey available on ubuntu AND macos? do we need to add it to installation requirements in README?

Comment (musnit) [reply parachain/scripts/make-snowbridge-image.sh:23]: or is this script not expected to be used locally?

Comment (musnit) [reply parachain/scripts/make-snowbridge-image.sh:23]: also which runtime will this image be built for?

File: parachain/src/chain_spec/rococo.rs [hunk 3]
```diff
@@ -61,7 +26,7 @@ where
 
 pub fn get_chain_spec(para_id: ParaId) -> ChainSpec {
 	let mut props = Properties::new();
-	props.insert("tokenSymbol".into(), "DEV".into());
+	props.insert("tokenSymbol".into(), "ROC".into());
```
Comment (musnit) [inline parachain/src/chain_spec/rococo.rs:29]: is this creating an additional ROC token on our parachain? or refering to the relay chain's token? i would rather call it SNOW or smth if its a new token

File: parachain/src/chain_spec/snowbridge.rs [hunk 4]
```diff
@@ -61,7 +26,7 @@ where
 
 pub fn get_chain_spec(para_id: ParaId) -> ChainSpec {
 	let mut props = Properties::new();
-	props.insert("tokenSymbol".into(), "DEV".into());
+	props.insert("tokenSymbol".into(), "DOT".into());
```
Comment (musnit) [inline parachain/src/chain_spec/snowbridge.rs:29]: same as above comment re ROC

File: parachain/src/chain_spec/snowbridge.rs [hunk 5]
```diff
@@ -151,46 +121,30 @@ fn testnet_genesis(
 				)
 			]
 		},
-		nft: NFTConfig {
+		nft: snowbridge_runtime::NFTConfig {
 			tokens: vec![]
 		},
-		verifier_lightclient: VerifierLightclientConfig {
```
Comment (musnit) [inline parachain/src/chain_spec/snowbridge.rs:157]: how does this get set now?

File: relayer/workers/beefyrelayer/store/store.go [hunk 6]
```diff
@@ -96,71 +100,78 @@ func NewDatabaseCmd(info *BeefyRelayInfo, cmdType CmdType, instructions map[stri
...
 	}
 }
 
-func PrepareDatabase(config *Config) (*gorm.DB, error) {
-	if len(config.DBPath) == 0 {
-		return nil, fmt.Errorf("invalid database path: %s", config.DBPath)
-	}
-	tmpDBFile, err := ioutil.TempFile("", config.DBPath)
+func (d *Database) Initialize() error {
+	tmpfile, err := ioutil.TempFile("", "beefy.*.db")
```
Comment (musnit) [inline relayer/workers/beefyrelayer/store/store.go:119]: 👍

File: relayer/workers/beefyrelayer/store/store.go [hunk 7]
```diff
@@ -96,71 +100,78 @@ func NewDatabaseCmd(info *BeefyRelayInfo, cmdType CmdType, instructions map[stri
...
+
+		d.log.Info("Shutting down DB")
+		sqlDB := d.DB.DB()
+		if sqlDB != nil {
+			err2 = sqlDB.Close()
+			if err2 != nil {
+				d.log.WithError(err2).Error("Unable to close DB connection")
+			}
+
+			err2 = os.Remove(d.Path)
```
Comment (musnit) [inline relayer/workers/beefyrelayer/store/store.go:156]: nice 👍 - if the relayer crashes and restarts without this, i presume it just erases/overwrites from scratch into a new file when it restarts?

File: relayer/workers/beefyrelayer/main.go [hunk 8]
```diff
@@ -135,18 +139,6 @@ func (worker *Worker) Start(ctx context.Context, eg *errgroup.Group) error {
 	return nil
 }
 
-func (worker *Worker) Stop() {
```
Comment (musnit) [inline relayer/workers/beefyrelayer/main.go:138]: we don't need this anymore?

File: relayer/workers/parachaincommitmentrelayer/beefy-listener.go [hunk 9]
```diff
@@ -55,6 +54,26 @@ func (li *BeefyListener) Start(ctx context.Context, eg *errgroup.Group) error {
 	}
 	li.beefyLightClient = beefyLightClientContract
 
+	// Fetch ParaId
```
Comment (musnit) [inline relayer/workers/parachaincommitmentrelayer/beefy-listener.go:57]: nice improvement - better than putting it in config as you say as we already have the endpoint in config

**Comment by musnit** [comment]
@vgeddes looks great, thanks! just some questions and minor comments to resolve and then can merge

File: ethereum/migrations/4_dump_address.js (hunk 1 above)
Comment (vgeddes) [reply ethereum/migrations/4_dump_address.js]: fixed. Renamed to 99_dump_address.js

File: relayer/workers/beefyrelayer/main.go (hunk 8 above)
Comment (vgeddes) [reply relayer/workers/beefyrelayer/main.go:138]: Nope. was never being called. Riz and I decided that workers shouldn't have a Stop function, and instead they should just clean up gracefully themselves after encountering an error.

**Pull request merged**